language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -671,4 +675,28 @@ public CompletableFuture<SemanticTokens> semanticTokensFull(SemanticTokensParams
             return new SemanticTokens(new ArrayList<>());
         });
     }
+
+    @Override
+    public CompletableFuture<List<InlayHint>> inlayHint(InlayHintParams params) {
+        return CompletableFutures.computeAsync((cancelChecker) -> {
+            try {
+                InlayHintContext context = ContextBuilder.buildInlayHintContext(
```

Comment:
We haven't considered the `range` in `InlayHintParams`. As per LSP, we only have to compute inlay hints for the visible section of the active editor (file)

Reply:
Let's create an issue to track that an improve